File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <iostream>
#include <string>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    
    stack<char> stk;
    for(char c:s){
        if(c=='(')  stk.push(c);
        else if(c==')'){
            if(stk.empty())    return false;
            stk.pop();
        }
    }
    if(!stk.empty())    answer=false;
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public bool solution(string s) 
    {
        bool answer = true;
        Stack<char> stk = new Stack<char>();
        foreach(var ch in s) {
            if(ch == ')') {
                if(stk.Count > 0 && stk.Peek() == '(') { stk.Pop(); }
                else { answer = false; break; }
            } else { stk.Push(ch); }
        }
        
        if(stk.Count > 0) { answer = false; }
        return answer;
    }
}